src/main/java/com/dabsquared/gitlabjenkins/util/TriggerUtil.java

```diff
@@ -0,0 +1,181 @@
+    private static final Logger LOGGER = Logger.getLogger(TriggerUtil.class.getName());
+    private static final Pattern TRIGGER_PATTERN = Pattern.compile("(triggers\\s{0,}\\{\\s{0,}gitlab\\s{0,}\\()" +
+        "([a-zA-Z: ,\"*]+)");
+
+    private TriggerUtil() {
+    }
+
+    public static <T extends WebHook> GitLabPushTrigger getFromJob(Item project, T webHook) {
+        if (project instanceof ParameterizedJobMixIn.ParameterizedJob) {
+            ParameterizedJobMixIn.ParameterizedJob job = (ParameterizedJobMixIn.ParameterizedJob) project;
+            for (Trigger trigger : job.getTriggers().values()) {
+                if (trigger instanceof GitLabPushTrigger) {
+                    return (GitLabPushTrigger) trigger;
+                }
+            }
+        }
+
+        if (project instanceof WorkflowMultiBranchProject) {
+            if (webHook == null) {
+                return null;
+            }
+
+            WorkflowMultiBranchProject multiBranchProject = (WorkflowMultiBranchProject) project;
+
+            FilePath dir;
+            Node node = getInstance();
+            Objects.requireNonNull(node, "Jenkins node not active");
+            if (multiBranchProject.getParent() instanceof TopLevelItem) {
+                FilePath baseWorkspace = node.getWorkspaceFor((TopLevelItem) multiBranchProject.getParent());
+                Objects.requireNonNull(baseWorkspace, "Could not get project workspace");
+                dir = getFilePathWithSuffix(baseWorkspace);
+            } else { // should not happen, but just in case:
+                dir = new FilePath(multiBranchProject.getRootDir());
+            }
+
+            try {
+                String branch = webHook.getEventSourceBranch();
+                // Uses same constant as in org.jenkinsci.plugins.workflow.multibranch.WorkflowBranchProjectFactory
+                FilePath scriptFile = dir.child("branches").child(branch).child("workspace").child("Jenkinsfile");
+                if (!scriptFile.absolutize()
+                    .getRemote()
+                    .replace('\\', '/')
+                    .startsWith(dir.absolutize().getRemote().replace('\\', '/') + '/')) { // TODO JENKINS-26838
+                    throw new IOException(scriptFile + " is not inside " + dir);
+                }
+                if (!scriptFile.exists()) {
+                    throw new AbortException(scriptFile + " not found");
+                }
+                String script = scriptFile.readToString().replace("\n", "");
+                GitLabPushTrigger trigger = parseTriggerFromPipeline(script);
+                trigger.setJob(((WorkflowMultiBranchProject) project).getItemByBranchName(branch));
+                return trigger;
+            } catch (Exception e) {
+                throw new RuntimeException(e);
+            }
+        }
+        return null;
+    }
+
+    private static GitLabPushTrigger parseTriggerFromPipeline(String script) throws AbortException {
```

Comment:
This is a bad idea. See discussion on the dev list, and reference to JENKINS-41929.